node/tests/GlideClient.test.ts

```diff
@@ -1501,6 +1501,65 @@ describe("GlideClient", () => {
         },
     );
 
+    function getRandomString(length: number) {
+        return Math.random()
+            .toString(36)
+            .substring(2, length + 2);
+    }
+
+    it.each([
+        [ProtocolVersion.RESP2, 5],
+        [ProtocolVersion.RESP2, 100],
+        [ProtocolVersion.RESP2, 1500],
+        [ProtocolVersion.RESP3, 5],
+        [ProtocolVersion.RESP3, 100],
+        [ProtocolVersion.RESP3, 1500],
+    ])(
+        "test inflight requests limit of %p with protocol %p",
+        async (protocol, inflightRequestsLimit) => {
+            const config = getClientConfigurationOption(
+                cluster.getAddresses(),
+                protocol,
+                { inflightRequestsLimit },
+            );
+            const client = await GlideClient.createClient(config);
+
+            try {
+                const key1 = `{nonexistinglist}:1-${getRandomString(10)}`;
+                const tasks: Promise<[GlideString, GlideString] | null>[] = [];
+
+                // Start inflightRequestsLimit blocking tasks
+                for (let i = 0; i < inflightRequestsLimit; i++) {
+                    tasks.push(client.blpop([key1], 0));
+                }
+
+                // This task should immediately fail due to reaching the limit
+                await expect(client.blpop([key1], 0)).rejects.toThrow(
+                    RequestError,
+                );
+
+                // Verify that all previous tasks are still pending
+                const timeoutPromise = new Promise((resolve) =>
```

Comment:
If I understood right, you are trying to check that all tasks are still pending, but what you check is that at least one of them is not resolved.
Meaning, if all of them resolved except one, the timeout will win since you use `Promise.all` which resolve when all resolve.
Look at the test i did to verify:
```javascript
const timeoutPromise = new Promise((resolve) => setTimeout(resolve, 100));
const tasks = [
    new Promise((resolve) => setTimeout(resolve, 50)),
    new Promise((resolve) => setTimeout(resolve, 75)),
    new Promise((resolve) => setTimeout(resolve, 400)),
];
const allTasksStatus = await Promise.race([
    Promise.all(tasks.map((task) => task.then(() => "resolved"))),
    timeoutPromise.then(() => "timeout"),
]);
console.log(allTasksStatus);
```
The result are 
```bash
[ec2-user@ip-10-0-0-24 ma-glide]$ node ./dist/index.js
timeout
```
Even tho two of them resolved before timeout the result are timeout because of the third.
Run it like that:
```javascript
const allTasksStatus = await Promise.race([
    ...tasks.map((task) => task.then(() => "resolved")),
    timeoutPromise.then(() => "pending"),
]);
```

Reply:
Changed to Promise.any instead of Promise.all.
Nice catch!